utils/eth1.go

```diff
@@ -346,6 +347,110 @@ func formatAmount(amount *big.Int, unit string, digits int, maxPreCommaDigitsBef
 	return template.HTML(fmt.Sprintf("<span%s>%s%s</span>", tooltip, preComma, displayUnit))
 }
 
+// NewFormat returns a new HTML-formatted string representing the given Wei big.Int amount, with the specified number of digits after the decimal point and number of digits hidden on mobile views.
+//
+// Parameters:
+//   - `amount` : the Wei(!) amount to be formatted.
+//   - `unit` : the unit in which the amount should be formatted. Supported values are ETH and GWei.
+//   - `digits` : the number of digits after the decimal point to be displayed.
+//   - `mobileDigitsHide` : the number of trailing decimal digits to hide on mobile views.
+//   - `maxPreCommaDigitsBeforeTrim` : the maximum number of digits before the decimal point that should be displayed before trimming excess decimal digits. If this is set to 0, no digits are trimmed.
+//
+// **NOTE** Every number is rounded down
+//
+// TODO Replace every formatAmount with this
+func NewFormat(amount *big.Int, unit string, digits int, mobileDigitsHide int, maxPreCommaDigitsBeforeTrim int) template.HTML {
+	// define display unit & digits used per unit max
+	var displayUnit string
+	var unitDigits int
+	// if more units are added, make sure to add them in the godoc
+	switch unit {
+	case "ETH":
+		displayUnit = " ETH"
+		unitDigits = 18
+	case "GWei":
+		displayUnit = " GWei"
+		unitDigits = 9
+	default:
+		displayUnit = " Wei"
+		unitDigits = 0
+	}
+
+	// Initialize preComma and postComma variables to "0"
+	preComma := "0"
+	postComma := "0"
+
+	if amount != nil {
```

Comment:
Passing nil as amount shows "0.0 ETH" and a tool-tip of "0.0".

Personally I think this function should not work if amount = nil so I would remove this if here and add a
```
if amount == nil {
	logger.Error("nil passed as amount to NewFormat")
	return template.HTML(`NA`)
}
```
at the very beginning of the function. Otherwise, errors may stay undetected and potentially wrong values are shown.

Thoughts?

Reply:
Good suggestion, was implemented without logging an error. I'm not 100% sure if passing a nil amount is always faulty behavior, once more designs have been implemented I will be able to evaluate this better.